node/src/cli.rs

```diff
@@ -96,54 +95,74 @@ pub struct Arguments {
     no_telemetry: bool,
 }
 
-impl Arguments {
-    /// Similar to [StructOpt::from_args] with additional information filled in by `version_info`.
-    pub fn from_args(version_info: &sc_cli::VersionInfo) -> Self {
-        let app = Arguments::clap()
-            .max_term_width(80)
-            .name(version_info.executable_name)
-            .author(version_info.author)
-            .about(version_info.description)
-            // We need to manually reset the `long_about` so that `structopt` does not take the
-            // code documentation of `Subcommand` for it.
-            .long_about("")
-            .settings(&[clap::AppSettings::UnifiedHelpMessage]);
-        Arguments::from_clap(&app.get_matches())
+impl SubstrateCli for Cli {
+    fn impl_name() -> &'static str {
+        "Radicle Registry Node"
     }
 
-    pub fn run_cmd(self) -> RunCmd {
-        // This does not panic if there are no required arguments which we statically know.
-        let mut run_cmd = RunCmd::from_iter_safe(vec![] as Vec<String>).unwrap();
+    fn impl_version() -> &'static str {
+        "ff.0"
+    }
 
-        let Arguments {
-            bootnodes,
-            data_path,
-            node_key,
-            node_key_file,
-            unsafe_rpc_external,
-            prometheus_external,
-            name,
-            ..
-        } = self;
-
-        run_cmd.network_config.bootnodes = bootnodes;
-        run_cmd.network_config.node_key_params.node_key = node_key;
-        run_cmd.network_config.node_key_params.node_key_file = node_key_file;
-        run_cmd.shared_params.base_path = data_path;
-        run_cmd.import_params.execution_strategies.execution =
-            Some(sc_cli::ExecutionStrategy::Both);
+    fn description() -> &'static str {
+        "Radicle Registry Node"
+    }
+
+    fn author() -> &'static str {
+        "Monadic GmbH"
+    }
+
+    fn support_url() -> &'static str {
+        "http://github.com/radicle-dev/radicle-registry/issues"
+    }
+
+    fn copyright_start_year() -> i32 {
+        2019
+    }
+
+    fn executable_name() -> &'static str {
+        "radicle-registry"
```

Comment:
```suggestion
        "radicle-registry-node"
```
Ideally we would get this from cargo with `env!` but I couldn’t find the right environment variable.

Reply:
Done: https://github.com/radicle-dev/radicle-registry/pull/465/commits/8d76d044507e56606313587ba9af5566bd0a48d2 :smiley: